client/client.go

```diff
@@ -0,0 +1,142 @@
+package client
+
+import (
+	"crypto/sha256"
+	"encoding/json"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path"
+	"search/indexer"
+	"search/server"
+	"search/util"
+	"sort"
+	"strconv"
+	"strings"
+)
+
+// Client stores the necessary information for a client.
+type Client struct {
+	mountPoint    string                      // Mount point for the client where all the files are stored
+	server        *server.Server              // The server that this client is connected to
+	indexer       *indexer.SecureIndexBuilder // The indexer for the client
+	lookupTable   map[string]string           // A map from docuemnt ids to actual filenames
+	reverseLookup map[string]string           // A map from actual filenames to document ids
+}
+
+// CreateClient instantiates a client connected to Server `s` with a
+// `clientNum`.
+func CreateClient(s *server.Server, clientNum int, mountPoint string) *Client {
+	c := new(Client)
+
+	c.server = s
+
+	// Calculates the master secret and sets up the indexer
+	h := sha256.New()
+	h.Write([]byte(strconv.Itoa(clientNum)))
+	serverKeyHalf := s.GetKeyHalf(clientNum)
+	ms := util.XorBytes(h.Sum(nil), serverKeyHalf, len(serverKeyHalf))
+	c.indexer = indexer.CreateSecureIndexBuilder(sha256.New, ms, s.GetSalts(), s.GetSize())
+
+	// Initializes the lookup table
+	// NOTE: Factor out and add decryption
+	c.lookupTable = make(map[string]string)
+	c.reverseLookup = make(map[string]string)
+	if tableContent, found := s.ReadLookupTable(); found {
```

Comment:
As we talked about today, I think it would be tough to synchronize the lookup table to all the clients this way.  We're better off using the full file name as the document ID, and updating that on the server whenever it changes.